lib/validators.js

```diff
@@ -32,6 +32,10 @@ exports.scope = Joi.string()
   .max(256)
   .regex(/^[a-zA-Z0-9 _\/.:]+$/);
 
+exports.redirectUri = Joi.string()
+  .max(256)
+  .regex(/^[a-zA-Z0-9\-_\/.:]+$/);
```

Comment:
The auth-server has a URI validator that we could copy for this, but I'm happy for that to be a follow-up.

Reply:
Did you mean https://github.com/mozilla/fxa-auth-server/blob/10e934f5fe71c464ffb0bcf1b71f5b61fd4afc47/lib/routes/validators.js#L176 ?

Reply:
Yes, and https://github.com/mozilla/fxa-auth-server/blob/10e934f5fe71c464ffb0bcf1b71f5b61fd4afc47/lib/routes/validators.js#L158 that uses it

Reply:
Hm at first glance it seems using that will break something like `testpilot-thing://oauth-redirect`

Reply:
Hrm, indeed; *shrug* I don't fell that strongly about it TBH so happy to just leave it for now :-)

Reply:
This pattern prohibits query strings in redirectUri, we have a few oauth relier with such redirectUris in China, is it an intentional limit or not?

Reply:
Filed as #547, thanks!